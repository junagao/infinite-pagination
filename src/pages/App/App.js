import React from 'react'
import { hot } from 'react-hot-loader/root'
import { List } from 'components'

const App = () => <List />

export default hot(App)
