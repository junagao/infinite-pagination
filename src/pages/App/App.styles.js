import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const GlobalStyles = css`
  html {
    margin: 0;
    border: 0;
    padding: 0;
    height: 100%;
    overflow-y: scroll;
    overflow-x: hidden;
  }
  *,
  *:before,
  *:after {
    box-sizing: inherit;
    font-family: inherit;
    margin: 0;
    padding: 0;
  }
  body {
    margin: 0;
    padding: 0;
    font-family: 'Roboto', sans-serif;
  }
  ::-webkit-scrollbar {
    width: 0px;
    height: 0px;
  }
  body::-webkit-scrollbar {
    width: 1.25rem;
    background-color: rgba(0, 0, 0, 0.05);
  }
  body::-webkit-scrollbar-thumb {
    background: linear-gradient(180deg, #f7971e, #ffd200);
    outline: 1px solid black;
    border-radius: 1rem;
  }
`

export const AppContainer = styled.div`
  text-align: center;
  margin: 0 auto;
`
