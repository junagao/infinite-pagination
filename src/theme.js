const theme = {
  colors: {
    background: '#fff',
    primary: '#111',
    text: { body: '#111', hover: '#f0f0f0' },
    disabled: '#ddd',
    error: '#BD271E',
  },
  fontSizes: {
    xl: '4rem',
    lg: '2rem',
    md: '1.5rem',
    sm: '1rem',
    xs: '0.75rem',
  },
  space: [
    0,
    '0.25rem',
    '0.5rem',
    '0.75rem',
    '1rem',
    '1.5rem',
    '2rem',
    '4rem',
    '6rem',
  ],
  borders: {
    none: 'none',
    thin: '1px solid',
  },
  radii: {
    none: 0,
    base: '0.25em',
    round: '99999em',
  },
}

export default theme
