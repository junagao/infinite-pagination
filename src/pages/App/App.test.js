import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import theme from '../../theme'
import App from './App'

describe('App', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
