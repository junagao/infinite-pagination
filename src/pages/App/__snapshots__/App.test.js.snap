// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
.emotion-2 {
  text-align: center;
  margin: 0 auto;
}

.emotion-0 {
  border: 0.25rem solid #f3f3f3;
  border-radius: 99999em;
  border-top: 0.25rem solid #777;
  width: 1.5rem;
  height: 1.5rem;
  -webkit-animation: spin 0.75s linear infinite;
  animation: spin 0.75s linear infinite;
  margin: 0 auto;
  margin-top: 2rem;
  margin-bottom: 4rem;
}

<div
  class="emotion-2 emotion-3"
>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    >
      <ul
        data-testid="list"
      />
      <div
        class="emotion-0 emotion-1"
        data-testid="spinner"
      />
    </div>
  </div>
</div>
`;
